pkg/cmd/spks.go

```diff
@@ -55,142 +54,145 @@ func SpksCMD() *cli.Command {
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
-			redisStandard, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[2], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
-			redisPremium, err := QueryPrometheus(ctx, v1api, prometheusQueryArr[3], logger, startOfToday)
-			if err != nil {
-				logger.Error(err, "Error querying Prometheus")
-				return err
-			}
+			now := time.Now().In(location)
+			// this variable is necessary to query Prometheus, with timerange [1d:1d] it returns data from 1 day up to midnight
+			startOfToday := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, location)
+			startYesterdayAbsolute := time.Date(now.Year(), now.Month(), now.Day()-1, 0, 0, 0, 0, location).In(time.UTC)
+			endYesterdayAbsolute := startYesterdayAbsolute.Add(24 * time.Hour)
 
-			odooClient := odoo.NewOdooAPIClient(c.Context, odooURL, odooOauthTokenURL, odooClientId, odooClientSecret, logger)
-
-			billingRecords := []odoo.OdooMeteredBillingRecord{
-				{
-					ProductID:     "appcat-spks-mariadb-standard",
-					InstanceID:    "mariadb-standard",
-					SalesOrder:    salesOrder,
-					UnitID:        UnitID,
-					ConsumedUnits: float64(mariadbStandard),
-					TimeRange: odoo.TimeRange{
-						From: startYesterdayAbsolute,
-						To:   endYesterdayAbsolute,
-					},
-				},
-				{
-					ProductID:     "appcat-spks-mariadb-premium",
-					InstanceID:    "mariadb-premium",
-					SalesOrder:    salesOrder,
-					UnitID:        UnitID,
-					ConsumedUnits: float64(mariadbPremium),
-					TimeRange: odoo.TimeRange{
-						From: startYesterdayAbsolute,
-						To:   endYesterdayAbsolute,
-					},
-				},
-				{
-					ProductID:     "appcat-spks-redis-standard",
-					InstanceID:    "redis-standard",
-					SalesOrder:    salesOrder,
-					UnitID:        UnitID,
-					ConsumedUnits: float64(redisStandard),
-					TimeRange: odoo.TimeRange{
-						From: startYesterdayAbsolute,
-						To:   endYesterdayAbsolute,
-					},
-				},
-				{
-					ProductID:     "appcat-spks-redis-premium",
-					InstanceID:    "redis-premium",
-					SalesOrder:    salesOrder,
-					UnitID:        UnitID,
-					ConsumedUnits: float64(redisPremium),
-					TimeRange: odoo.TimeRange{
-						From: startYesterdayAbsolute,
-						To:   endYesterdayAbsolute,
-					},
-				},
-			}
+			odooClient := odoo.NewOdooAPIClient(c.Context, odooURL, odooOauthTokenURL, odooClientId, odooClientSecret, logger, allMetrics["odooMetrics"])
 
 			ticker := time.NewTicker(24 * time.Hour)
-			sigint := make(chan os.Signal, 1)
-			signal.Notify(sigint, os.Interrupt)
 
-			// ensure first data export after restart
-			// I want to avoid situation when data is not exported because of constant restarts
-			err = odooClient.SendData(billingRecords)
-			if err != nil {
-				logger.Error(err, "can't export data to Odoo")
-				return err
-			}
+			firstTime := make(chan bool, 1)
```

Comment:
You could just call `runSPKSBilling()` here instead of creating an additional channel.

Reply:
actually this is leftover from way another approach :D I have no idea why it survived